pyhoma/client.py

```diff
@@ -253,14 +299,19 @@ async def get_devices(self, refresh: bool = False) -> list[Device]:
     )
     async def get_gateways(self, refresh: bool = False) -> list[Gateway]:
         """
-        List gateways
+        Get every gateways of a connected user setup
+        Per-session rate-limit : 1 calls per 1d period for this particular operation (bulk-load)
         """
         if self.gateways and not refresh:
             return self.gateways
 
         response = await self.__get("setup/gateways")
         gateways = [Gateway(**g) for g in humps.decamelize(response)]
+
+        # Cache response
         self.gateways = gateways
+        if self.setup:
+            self.setup.gateways = gateways
 
         return gateways
 
```

Comment:
What about:

```python
@property
def setup(self):
  if  self.setup:
    self.setup.gateways = self.gateways
    self.setup.devices = self.devices
  return setup
```

Reply:
This won't work, since we need to have the latest cached information. It can happen that setup has newer information than the information in self.devices or self.gateways.